[Woo Jian Zhe] iP

jzwoo opened the pull request.

Commit:
Add Gradle support

Commit:
Implemented Level-1

Commit:
Implemented Level-2

Commit:
Implemented Level-3

Commit:
Implemented level-3 with Task class

Commit:
Implemented Level-4

Commit:
Tried the Automated Text UI Testing

Commit:
Implemented Level-5

Commit:
Implemented Level-6 delete feature

Commit:
Implemented Level-7 with trivial handling of exceptions

Commit:
Implemented Level-8 without stretch

Commit:
Merge branch 'branch-level-7'

Commit:
Implemented Level-8 and fixed merge conflicts

Commit:
Made IP more OOP

Commit:
Added JUnit test class

Commit:
Added java docs to almost all methods

Commit:
Complied with coding standard

Commit:
Implemented Level 9 find feature

Commit:
Merged A-CodingStandard with master and fixed merge conflicts

Commit:
Merged Level-9 with master

Commit:
Cleared up the code and added JavaDocs to all classes

Commit:
Updated Junit test class

Commit:
Minor changes

Commit:
Merge commit '3b19ba1' into branch-A-Gradle

Commit:
Set up gradle

Commit:
Implemented GUI

onglijin commented on src/test/TaskListTest.java:
```diff
@@ -0,0 +1,52 @@
+class TaskListTest {
+
+    String home = System.getProperty("user.home");
+    File testFile = new File(home + "/ip/src/main/java/Data/TestFile.txt");
```
I think perhaps you could abstract ```"/ip/src/main/java/Data/TestFile.txt"``` to avoid magic strings?

onglijin commented on src/test/ParserTest.java:
```diff
@@ -0,0 +1,150 @@
+class ParserTest {
+
+    @Test
+    void parse_listCommand_1() {
```
Perhaps the naming of this method can be changed? The usual convention is ```whatIsBeingTested_descriptionOfTestInputs_expectedOutcome```.

onglijin commented on src/main/java/UI.java:
```diff
@@ -0,0 +1,152 @@
+
+    /**
+     * Reads the user input and communicates with the storage and parser on what to do next.
+     * @throws IOException If the storage cannot find the file containing the saved tasks.
```
I think perhaps a space could be added in between these 2 sentences?

onglijin commented on src/main/java/UI.java:
```diff
@@ -0,0 +1,152 @@
+
+                        this.storage.markDone(index);
+
+                        System.out.println("Nice! I've marked this task as done:" + "\n" + "  " + this.storage.getListTask(index).toString());
```
Perharps you could break at the ```+``` as the statement exceeds 110 characters (soft limit) / 120 characters (hard limit)?

onglijin commented on src/main/java/UI.java:
```diff
@@ -0,0 +1,152 @@
+        }
+    }
+
+    private boolean isListCommand(int num) {
+        return num == 1;
+    }
+
+    private boolean isDoneCommand(int num) {
+        return num == 2;
+    }
+
+    private boolean isDeleteCommand(int num) {
+        return num == 3;
+    }
+
+    private boolean isFindCommand(int num) {
+        return num == 4;
+    }
```
Perhaps a switch case could be used here?

onglijin commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,134 @@
+    private static Task convertToTask(String line) {
+        if (line.startsWith("[T]", 2)) {
+            //Is a todo task
+            String[] parts = line.split(" ", 2);
+            if (line.contains("[✘]")) {
+                return new Task(parts[1]);
+            } else {
+                return new Task(parts[1]).markDone();
```
Perhaps the method could abstracted into another method to reduce the length of the method ```convertToTask```?

onglijin reviewed (commented):
Almost LGTM. Remember to leave an empty line between description and parameters for method comments and line warp at appropriate places when the code gets too long. Overall, well done!

Commit:
Integrated duke with the GUI

Commit:
Refactor code to improve readability.

Commit:
Add assertions

Commit:
Refactor code according to Code Quality guide

jzwoo committed:
Merge pull request #1 from jzwoo/branch-A-Assertions

Add assertions

jzwoo committed:
Merge branch 'master' into branch-A-CodeQuality

jzwoo committed:
Merge pull request #2 from jzwoo/branch-A-CodeQuality

Refactor code according to Code Quality guide

Commit:
Implement C-Update

The current Duke chat bot does not allow for the Tasks to be edited after adding into the task list.

Implementing C-Update allows users of Duke bot to replace the description, time, date or the task entirely, making it more user friendly.

Commit:
Fix style errors according to checkstyle

Commit:
Use GitHub Actions to set up CI

Commit:
Improve Duke Gui

GUI is mandane and not user-friendly.

A more pleasing GUI can improve user experience.

Let's,
* increase font size
* increase window size
* adjust dialog bubble to fit messages
* add a background image
* add a speech bubble to dialog

Commit:
Add user guide

Commit:
Add Ui.png into docs folder

Commit:
Refactor code and add abort command for update

Some functions are long and and update command does not allow undo.

Improves the code quality and user now has the ability to abort the update.

Commit:
Update user guide

Commit:
Update GUI

Commit:
Update JavaDocs and user guide
